urllib3/response.py

```diff
@@ -327,9 +360,15 @@ def read(self, amt=None, decode_content=None, cache_content=False):
                     # no harm in redundantly calling close.
                     self._fp.close()
                     flush_decoder = True
+                    if (self.strict_content_length and self.length not in (0, None)):
+                        # httplib requires an iterator/string instead of count of bytes
+                        data = [0 for x in range(self._fp_bytes_read)]
```

Comment:
This is awkward. This error is used everywhere else this issue occurs though, so it feels off to implement another error to avoid the need of this iterator.